src/y-indexeddb.js

```diff
@@ -24,7 +24,6 @@ export const fetchUpdates = (idbPersistence, beforeApplyUpdatesCallback = () =>
       afterApplyUpdatesCallback(updatesStore)
     }
   })
-    .then(() => idb.getLastKey(updatesStore).then(lastKey => { idbPersistence._dbref = lastKey + 1 }))
```

Comment:
Thanks for taking the time to optimize this library!

`_dbref` is intentionally only updated in `fetchUpdates`.

When y-indexeddb is used in multiple browser tabs, every tab is responsible for keeping in-sync with the database. We do broadcast updates using `broadcastchannel`, but there is no guarantee that it will always work (there are some edge-cases when broadcastchannel it might fail - e.g. when the browser tab is terminated). If one update is not properly broadcasted to all tabs, then the tab is no longer synced with the database. By simply bumping `_dbref` after storing an update, we also don't know anymore whether we missed an update.

In line 38 (see `PREFERRED_TRIM_SIZE`)  we "concatenate" all updates into a single entry. We store the current state of the Yjs document and remove all other entries. If that document is not in-sync with the database, we will unintentionally discard some entries from other browser-tabs.

So I'm afraid that we can't apply this optimization..




There is no way to guarantee that

Reply:
Got it! I hadn't thought about multiple tabs. Thanks so much for the explanation.